store/store.go

```diff
@@ -40,33 +53,50 @@ func OpenStore(logger tmlog.Logger, path string, options *badger.Options) (*Stor
 		return nil, err
 	}
 
-	ok := IsInit(logger, path)
+	ok := IsInit(logger, path, InitOptions{
+		NeedDataStore:   options.HasDataStore,
+		NeedEVMKeyStore: options.HasEVMKeyStore,
+		NeedP2PKeyStore: options.HasP2PKeyStore,
+	})
 	if !ok {
 		flock.Unlock() //nolint: errcheck
 		return nil, ErrNotInited
 	}
 
-	ds, err := badger.NewDatastore(dataPath(path), options)
-	if err != nil {
-		return nil, fmt.Errorf("can't open Badger Datastore: %w", err)
+	var ds *badger.Datastore
+	if options.HasDataStore {
+		if options.BadgerOptions == nil {
+			return nil, fmt.Errorf("badger data store options needed to open the store")
+		}
+		ds, err = badger.NewDatastore(dataPath(path), options.BadgerOptions)
+		if err != nil {
+			return nil, fmt.Errorf("can't open Badger Datastore: %w", err)
+		}
+	}
```

Comment:
there was recently a mem leak issue in node for using a slightly incorrect config. I don't think we'll have the same issues since this is just a keystore, but it might be something that we want to watch for casually

Reply:
We're using the geth keystore, I don't think this should affect us

Reply:
Just checked the leak, it affects use :dancers: I will fix, thanks a lot for pointing it

Reply:
https://github.com/celestiaorg/orchestrator-relayer/issues/269